.travis.yml

```diff
@@ -1,41 +1,160 @@
 
-  matrix:
-    - GHCVERSION=ghc865 STRICT=false TRACING=false
-#     - GHCVERSION=ghcjs
-# 
-# matrix:
-#   allow_failures:
-#     - env: GHCVERSION=ghcjs
+#  2020-05-25: NOTE: We still yet to build this in `master`
+jobs:
+  allow_failures:
+    - env: GHCVERSION=ghcjs
+    - env: GHCVERSION=ghc8101
+  #  2020-05-26: NOTE: Discard macOS GHCJS build + -> CI loop becomes ~30-40 minutes shorter
+  exclude:
+    - os: osx
+      env: GHCVERSION=ghcjs
+    - os: osx
+      env: GHCVERSION=ghc8101
+    - os: osx
+      env: GHCVERSION=ghc865
+    # NOTE: Discard Linux GHC 8.8.3 in favour of macOS GHC 8.8.3, Nixpkgs channel is the same and Nix package code is the same, since we test other Linux deployments and since 8.8.3 builds on macOS - we can be sure it builds inside Nix in Linux also. So lets optimize number of CI builds
+    - os: linux
+      env: GHCVERSION=ghc883
 
 before_script:
   - sudo mount -o remount,exec,size=4G,mode=755 /run/user || true
   - sudo mkdir -p /etc/nix
   - echo "trusted-users = root $USER" | sudo tee -a /etc/nix/nix.conf
+  # NOTE: macOS service restart
   - sudo launchctl kickstart -k system/org.nixos.nix-daemon || true
 
 script:
-  - nix-env -iA cachix -f https://github.com/NixOS/nixpkgs/tarball/db557aab7b690f5e0e3348459f2e4dc8fd0d9298
-  - cachix use hnix
-  - if [ ! "$CACHIX_SIGNING_KEY" = "" ]; then cachix push hnix --watch-store& fi
-  - if [ ! "$CACHIX_SIGNING_KEY" = "" ]; then ./build.sh | cachix push hnix; else ./build.sh; fi
+  #
+  #
+  # NOTE: Install Cachix client using Nix:
+  - nix-env -iA cachix -f https://cachix.org/api/v1/install
+  - cachix use "$name"
+  # NOTE: If key is set - use Cachix push, else - proceed without it
+  - if [ ! "$CACHIX_SIGNING_KEY" = "" ]; then cachix push "$name" --watch-store& fi
+  # NOTE: Brush timeout for previous daemon to start
+  - sleep 2
+  #
+  #
+  # NOTE: Normal GHC build
+  - if [ ! "$GHCVERSION" = 'ghcjs' ]; then ./build.sh; fi
+  #
+  #
+  # NOTE: GHCJS build
+  # HACK: Travis terminates GHCJS because of very huge log,
+  # so `SILENT` mode for it was created
+  # and travis_wait 50 to wait on no outputs (otherwise Travis terminates build in 10 minutes as stale)
+  # and `bash` wrapper so Travis parses `travis_wait` => `if; then; fi` line
+  - travis_wait 50 bash -c 'if [ "$GHCVERSION" = "ghcjs" ]; then ./build.sh; fi'
+  # NOTE: For GHCJS dump the last $ghcjsLogTailLength lines into CI out to see
+  # Since build runs inside `travis_wait` - it was impossible to output log from it
+  - if [ "$GHCVERSION" = "ghcjs" ]; then tail -n "$ghcjsLogTailLength" "$ghcjsTmpLogFile" && rm "$ghcjsTmpLogFile"; fi
+  #
+  #
+
+# NOTE: Travis cache allows near fast cache loading of Nix pkgs store,
+# Since the Cachix does not cache PRs - Travis cache is great because it caches PR branch builds
+# During devepment of this PR I had ~40GB cache on the PR branch.
+cache:
+  directories:
+    - /nix/store
 
+# NOTE: Track the commits on this repo branches + cron rechecks build
```

Comment:
What's that `cron` reference about? Is it related to the `pending` branch???

Reply:
`cron` is a well-known scheduling utility. It essentially was/is shorter synonym to _periodical_ or _on schedule_, synonym may got out of fashion just a couple of years ago due to `systemd timers` trying to replace `cron`, but `cron` is still widely used and `cron job` is a shortcut to `job that runs on periodical schedule`.

Reply:
Thanks @Anton-Latukha. I know what `cron` is. What's the purpose of the comment though? Is this some kind of `TODO`???

Reply:
This is a declarative statement that explains what the Travis YAML `branch:` key really does, since its name is so obscure.

Reply:
Thanks for the explanation @Anton-Latukha! :)

I thought you were possibly hinting at [Travis' cron job feature](https://docs.travis-ci.com/user/cron-jobs/), but apparently that's not it.